src/record/record.rs

```diff
@@ -309,10 +288,25 @@ impl Header {
         bincode::serialized_size(&self).expect("calc record serialized size")
     }
 
-    fn update_checksum(&mut self) -> bincode::Result<()> {
+    pub(crate) fn set_offset_checksum(&mut self, blob_offset: u64, header_checksum: u32) {
+        self.blob_offset = blob_offset;
+        self.header_checksum = 0;
+        debug_assert_eq!(self.crc32().expect("debug assert crc32"), header_checksum);
+        self.header_checksum = header_checksum;
+    }
+
+    pub(super) const fn blob_offset_offset(len: usize) -> usize {
```

Comment:
Please, add comment, that this function calculates offset to the `offset` field in serialized header. And that `len` is the length of serialized header in bytes